presto-main/src/main/java/com/facebook/presto/sql/planner/GroupingOperationRewriter.java

```diff
@@ -39,24 +39,18 @@
 import static com.google.common.base.Preconditions.checkState;
 import static java.util.Objects.requireNonNull;
 
-public class GroupingOperationRewriter
+public final class GroupingOperationRewriter
 {
-    private final QuerySpecification queryNode;
-    private final Analysis analysis;
-    private final Metadata metadata;
-    private final Optional<Symbol> groupIdSymbol;
+    private GroupingOperationRewriter() {}
 
-    public GroupingOperationRewriter(QuerySpecification queryNode, Analysis analysis, Metadata metadata, Optional<Symbol> groupIdSymbol)
+    public static Expression rewriteGroupingOperation(GroupingOperation expression, QuerySpecification queryNode, Analysis analysis, Metadata metadata, Optional<Symbol> groupIdSymbol)
     {
-        this.queryNode = requireNonNull(queryNode, "node is null");
-        this.analysis = requireNonNull(analysis, "analysis is null");
-        this.metadata = requireNonNull(metadata, "metadata is null");
-        this.groupIdSymbol = requireNonNull(groupIdSymbol, "groupIdSymbol is null");
-    }
+        requireNonNull(queryNode, "node is null");
+        requireNonNull(analysis, "analysis is null");
+        requireNonNull(metadata, "metadata is null");
+        requireNonNull(groupIdSymbol, "groupIdSymbol is null");
 
-    public Expression rewrite(GroupingOperation expression)
-    {
-        Expression rewrittenExpression = rewriteGroupingOperationToExpression(expression, queryNode);
+        Expression rewrittenExpression = rewriteGroupingOperationToExpression(expression, queryNode, analysis, groupIdSymbol);
         if (rewrittenExpression instanceof FunctionCall) {
```

Comment:
no need for this `if`. This code can be part of `rewriteGroupingOperationToExpression`